src/main/java/com/bytecoders/pharmaid/service/SharedPermissionService.java

```diff
@@ -0,0 +1,153 @@
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+/**
+ * Has methods to create, accept, deny, revoke permission share requests. Logic to ensure requester
+ * is current user yet be added.
+ */
+@Slf4j
+@Service
+public class SharedPermissionService {
+
+  private static final int PENDING = 0;
+  private static final int ACCEPT = 1;
+  private static final int DENY = 2;
+
+  private static final int VIEW = 0;
+  private static final int EDIT = 1;
+
+
+  @Autowired private SharedPermissionRepository sharedPermissionRepository;
+
+  @Autowired private UserRepository userRepository;
+
+  /**
+   * Creates a new sharing request.
+   *
+   * @param ownerId Id of owner.
+   * @param permissionType 0 if view, 1 if edit.
+   * @return SharedPermission object.
+   */
+  public SharedPermission createSharingRequest(
+      String requesterId, String ownerId, Integer permissionType) {
+
+    User owner =
+        userRepository
+            .findById(ownerId)
+            .orElseThrow(
+                () ->
+                    new ResponseStatusException(
+                        HttpStatus.NOT_FOUND, "User not found with ID: " + ownerId));
+
+    if (ownerId.equals(requesterId)) {
+      throw new IllegalArgumentException("Cannot create shared permission with self");
+    }
+
+    if (!(permissionType == VIEW || permissionType == EDIT)) {
+      throw new IllegalArgumentException("Permission Type should be view or edit");
+    }
+
+    User requester =
+        userRepository
+            .findById(requesterId)
+            .orElseThrow(
+                () ->
+                    new ResponseStatusException(
+                        HttpStatus.NOT_FOUND, "User not found with ID: " + requesterId));
+
+    // Checks if permission already exists and return it if found.
+    Optional<SharedPermission> existingPermission =
+        sharedPermissionRepository.findByOwnerRequesterPermissionStatus(
+            owner, requester, permissionType, PENDING
+            );
+
+    if (existingPermission.isPresent()) {
+      return existingPermission.get();
+    }
+
+    SharedPermission permission = new SharedPermission();
+    permission.setOwner(owner);
+    permission.setRequester(requester);
+    permission.setPermissionType(permissionType);
+    permission.setStatus(PENDING);
+
+    return sharedPermissionRepository.save(permission);
+  }
+
+  /**
+   * Accepts/Denies a share request.
+   *
+   * @param ownerId owner id
+   * @param permissionId id of permission to be accepted/denied
+   * @param accept 1 if accept, 2 if deny
+   * @return SharedPermission object
+   */
+  public SharedPermission acceptDenySharingRequest(
+      String ownerId, String permissionId, int accept) {
```

Comment:
nit: might make more sense to change `int accept` to something like `int status`